### DISCUSSION TITLE
Dynamically adding rules in a running system

### DISCUSSION THREAD

**Discussion by rishijain300900** [hyperjumptech/grule-rule-engine#255] [Q&A] [answered]
Hi
Can a running system have a new rule added dynamically. If so how.
For reference in drools we have the following:
https://stackoverflow.com/questions/9422106/drools-ability-to-define-rules-at-runtime
Do we have something similar in grule

**Comment by newm4n** [comment, marked as answer]
In Grools, its not that dynamic. 
You can't modify the rule set (`KnowledgeBase`) of the running engine.
You have to modify the rule set in `KnowledgeLibrary`, obtain another instance of `KnowledgeBase` from the library and run again.
